.github/groom/tests/test_patch_policy.py

```
@@ -191,6 +191,127 @@ def test_case_insensitive_on_case_insensitive_runners(self):
+        )
+        self.assertEqual(
+            policy.parse_raw_z(data),
+            [("100644", "100644", "src/foo.py"), ("000000", "120000", "suites/link")],
+        )
+
+    def test_parse_raw_z_empty_and_malformed(self):
+        self.assertEqual(policy.parse_raw_z(b""), [])
+        # A field where a meta record must sit but doesn't parse fails LOUD (the
+        # gate runs under `set -euo pipefail`, so a raise fails closed, not open):
+        # name-only-shaped input (the old producer) and an unpaired field both die.
+        with self.assertRaises(ValueError):
```

Ruff (0.16.2)

[warning] Use `pytest.raises` instead of unittest-style `assertRaises`

Replace `assertRaises` with `pytest.raises`

(PT027)